runelite-client/src/main/java/net/runelite/client/plugins/jewelrycount/JewelryCountOverlay.java

```diff
@@ -58,33 +58,74 @@ public Dimension render(Graphics2D graphics)
 
 		Widget inventory = client.getWidget(WidgetInfo.INVENTORY);
 
-		if (inventory == null || inventory.isHidden())
+		if (inventory == null)
 		{
 			return null;
 		}
 
-		for (WidgetItem item : inventory.getWidgetItems())
+		if (!inventory.isHidden())
 		{
-			JewelryCharges charges = JewelryCharges.getCharges(item.getId());
+			for (WidgetItem item : inventory.getWidgetItems())
+			{
+				JewelryCharges charges = JewelryCharges.getCharges(item.getId());
+
+				if (charges == null)
+				{
+					continue;
+				}
+
+				Rectangle slotBounds = item.getCanvasBounds();
```

Comment:
Deduplicate this logic and the drawing logic from below by splitting it into a function